### PR TITLE
[curl] Update to 7.77.0

### LINKED ISSUE #16119: [curl] update to <curl-7_75_0>
Library name: curl

New version number: 7.75.0

Other information that may be useful (release notes, etc...)
https://curl.se/changes.html

### PR DESCRIPTION
**Pull request by past-due**
Update `curl` port to 7.75.0: https://curl.se/changes.html

- What does your PR fix? Fixes #16119

TODO:
- [ ] Create new version of `0012-fix-dependency-idn2.patch`, if needed. (Possibly incorporating the `find_library` tweaks from the original? But taking advantage of the new `USE_LIBIDN2` built-in curl CMake option.)

### CODE REVIEW

File: ports/curl/0012-fix-dependency-idn2.patch [hunk 1]
```diff
@@ -1,51 +0,0 @@
-diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (JonLiu1993) [inline ports/curl/0012-fix-dependency-idn2.patch:1]: Why delete this patch?

**Comment by JonLiu1993** [comment]
@past-due ,Can you handle these conflicts?

File: ports/curl/0012-fix-dependency-idn2.patch (hunk 1 above)
Comment (past-due) [reply ports/curl/0012-fix-dependency-idn2.patch:1]: As of curl 7.75.0, curl now provides its own `USE_LIBIDN2` CMake option. An adapted patch may still be needed to handle the `find_library` tweaks - have not had a full chance to dig into this.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
@past-due Let us know when you're ready for review/merge by unmarking this as a draft :)

@JonLiu1993 please do not promote other people's PRs from draft to "ready for review"

**Comment by JonLiu1993** [comment]
@past-due ,Can this pr be reviewed now?

**Comment by past-due** [comment]
> @past-due ,Can this pr be reviewed now?

@JonLiu1993 I apologize for the delay - I have not had a chance to address or investigate the "TODO" listed in the first comment. If that is not an issue for anyone, I have no issues with it being reviewed, but at this point I'm uncertain of the impact.

**Comment by JonLiu1993** [comment]
@past-due ,Could you handle these conflicting files?

**Comment by dg0yt** [comment]
FTR, upstream curl is at 7.76.1, with two CVEs fixed since 7.75.0

**Comment by JonLiu1993** [comment]
@past-due ,Can you please resolve the conflicts against master?

**Comment by past-due** [comment]
@JonLiu1993 I have rebased and resolved conflicts with the master branch.

curl is now up to 7.77, but there appear to be more patches that need fixing, and I'm not sure when I'll have a chance to address.

Also, as noted above:
> Testing of the idn2 feature of the port is what's needed. I'll mark this as "ready for review", but I have not had a thorough chance to investigate the impact to that port feature. Please feel free to push whatever changes are needed.

**Comment by past-due** [comment]
I pushed an update to cURL 7.77.0, and some (attempts) at adjusting patches. Waiting for CI.

**Comment by dg0yt** [comment]
gdal failing on osx? Well, that is part of gdall being broken on Linux (#9068), and I have another curl patch pending for osx which is related, #17790. It is reviewed, but got a merge conflict before being merged... I will take a look if I can align this with your change.

File: ports/curl/0012-fix-dependency-idn2.patch (hunk 1 above)
Comment (dg0yt) [reply ports/curl/0012-fix-dependency-idn2.patch:1]: idn2 needs iconv, but curl doesn't link this when testing.

**Comment by dg0yt** [comment]
I will refresh #17790 independent of this PR, and that start with this PR separately. There a few more issues with port curl at the moment, first of all a mismatch of non-htp/HTTP_ONLY.

File: ports/curl/0012-fix-dependency-idn2.patch (hunk 1 above)
Comment (dg0yt) [reply ports/curl/0012-fix-dependency-idn2.patch:1]: I'll take this to #18971. I even had a look at idn2 recently, for mingw.

**Comment by JonLiu1993** [comment]
@past-due ,Can you please resolve the conflicts against master?

**Comment by dg0yt** [comment]
> @past-due ,Can you please resolve the conflicts against master?

Actually I already did this. I just wait for the current CI run of  #18971 to complete before pushing the cherry-picked commits. And I have pending work with regard to libidn2.

**Comment by PhoebeHui** [comment]
Closing in favor of https://github.com/microsoft/vcpkg/pull/18971.

**Pull request closed without merging**